File: internal/projects/lobby.go
package projects

import "path/filepath"

// ShouldShowLobby es true si hay que mostrar la pantalla de inicio: lista vacía,
// cwd no es git, o el toplevel del cwd no está entre los proyectos registrados.
func ShouldShowLobby(seed string, paths []string) bool {
	if len(paths) == 0 {
		return true
	}
	absSeed, err := filepath.Abs(seed)
	if err != nil {
		return true
	}
	top, err := GitTopLevel(absSeed)
	if err != nil {
		return true
	}
	top = filepath.Clean(top)
	for _, p := range paths {
		if filepath.Clean(p) == top {
			return false
		}
	}
	return true
}

File: src/projects/lobby.ts
import path from "node:path";
import { gitTopLevel } from "./gitMeta.js";

export async function shouldShowLobby(
  seed: string,
  paths: string[]
): Promise<boolean> {
  if (paths.length === 0) {
    return true;
  }
  let absSeed: string;
  try {
    absSeed = path.resolve(seed);
  } catch {
    return true;
  }
  let top: string;
  try {
    top = await gitTopLevel(absSeed);
  } catch {
    return true;
  }
  const cleanTop = path.normalize(top);
  for (const p of paths) {
    if (path.normalize(p) === cleanTop) {
      return false;
    }
  }
  return true;
}
